README.md

```diff
@@ -1,2 +1,82 @@
 # sonar-pipeline
-Pipeline to deploy sonarqube server to kubernetes
+Pipeline to deploy sonarqube server to kubernetes. Currently it only supports updating sonarqube's image. Other configuration requires manual update via helm.
+
+## First Time Deployment
+To simplify the deployment, we are using helm, the package manager for k8s, to bundle all we need for sonar and deploy them all in once. Example `values.yaml` can be found under `config-example` folder. The actual config files are stored somewhere else secretly.
+- Prerequisite:
+    1. install helm client on your dev machine: `brew install kubernetes-helm`
+    2. init helm in k8s cluster: https://docs.helm.sh/using_helm
+- Deployment:
+    1. Switch to the desired context
+    ```bash
+    kubectl config use-context desired-cluster-context
+    ```
+
+    2. Deploy beta sonar using sonarqube chart
+    ```bash
+    helm install --name beta -f beta-values.yaml stable/sonarqube
+    ```
+
+    3. Deploy prod sonar using sonarqube chart
+    ```bash
+    helm install --name prod -f values.yaml stable/sonarqube
+    ```
+
+    After deployment, you should be able to see such a set of resources created for sonarqube, e.g. for beta:
+
+    ```bash
+    ==> v1/Service
+    beta-postgresql  
+    beta-sonarqube   
+
+    ==> v1beta1/Deployment
+    beta-postgresql  
+    beta-sonarqube   
+
+    ==> v1beta1/Ingress
+    beta-sonarqube
+
+    ==> v1/Pod(related)
+    beta-postgresql
+    beta-sonarqube
+
+    ==> v1/Secret
+    beta-postgresql
+
+    ==> v1/ConfigMap
+    beta-sonarqube-config         
+    beta-sonarqube-install-plugins  
+    beta-sonarqube-startup          
+    beta-sonarqube-tests            
+
+    ==> v1/PersistentVolumeClaim
+    beta-postgresql   
+    beta-sonarqube
+    ```
+
+- Manual update in sonar UI:
+
+    For the following config, it's not configurable during boot time. Need to do it after sonar is up and running.
+  1. Go to **${your-sonar-host}** and log in as `admin` with default password `admin`.
```

Comment:
oh boy

Reply:
I know... They put it as a future improvement. https://github.com/SonarSource/docker-sonarqube/blob/master/recipes.md